### PR TITLE
Cleanup unused CoreGroup methods

### LINKED ISSUE #645: Cleanup unused CoreGroup methods
Quite a few methods are marked as test-only because they are no longer used and should be removed along with the tests.

### PR DESCRIPTION
**Pull request by raphaelrobert**
Fixes #645.

This PR does the following:

 - Remove test functions from the CoreGroup that are not useful along with the corresponding tests.
 - Moves code in `core_group/mod.rs`
 - Refactor unnecessary errors
 - Minor renaming

### CODE REVIEW

File: openmls/src/framing/private_message.rs [hunk 1]
```diff
@@ -129,8 +129,7 @@ impl PrivateMessage {
         )
     }
 
-    /// Internal function to encrypt content. The extra message header is only used
-    /// for tests. Otherwise, the data from the given `AuthenticatedContent` is used.
+    /// Internal function to encrypt content.
```
Comment (kkohbrok) [inline openmls/src/framing/private_message.rs:132]: Is there a particular reason you're removing the remark about the extra header here?

**Review by kkohbrok — COMMENTED** [review]
Thanks! One question, otherwise this looks good!

File: openmls/src/framing/private_message.rs (hunk 1 above)
Comment (raphaelrobert) [reply openmls/src/framing/private_message.rs:132]: It seemed outdated and no longer necessary.

Comment (kkohbrok) [reply openmls/src/framing/private_message.rs:132]: It does (still) describe what's going on in the code and I would say it's useful for anyone who is looking to understand what this function does.

**Review by kkohbrok — APPROVED** [review]

**Pull request merged**